main.js

```diff
@@ -834,6 +834,20 @@ function extendChannel(nodeID, comClass, valueId) {
 
     // Create channel
     if (objects[channelID]) {
+        // channel exists - make sure we don't have duplicate states
+        if (!nodes[nodeID].ready) {
```

Comment:
Whats the purpose of this check? Would it make sense to merge this cleanup with the already existing cleanup code after the scan is completed?
==> https://github.com/ioBroker/ioBroker.zwave/blob/7bc7f9d57be830bee442639bf95e03b97059cc01/main.js#L1099

Reply:
To do that cleanup together with the existing one was also my initial idea. However the only possibility I found to identify which states are actually obsolete is when the correct labels are coming in. And in the cleanup function I don't have that info. I have not found any other place to identify which labels/states are recent or not.
The additional check is just performance improvement as I noticed that this extendChannel function is called for *every* value update (despite the naming). So I need to make sure that any kind of cleanup is just done during the initial phase as it causes some load. I would have hoped any kind of channel/state update is separated from the value updates but it is not...

Reply:
Ok that makes sense